src/puya/ir/optimize/control_op_simplification.py

```diff
@@ -190,13 +191,16 @@ def remove_target(parent: models.BasicBlock, to_remove: models.BasicBlock) -> No
                 if default_block not in block.terminator.targets():
                     remove_target(block, default_block)
                 # add this block as a predecessor to any new targets
+                # and update relevant phi args in successor
                 for succ in unique(block.successors):
                     if block not in succ.predecessors:
```

Comment:
If `block` is in `succ.predecessors` what should happen with Phi arguments in `succ`? Perhaps this optimization should only occur if all successor phi argument values for `block` match  phi argument values for `default_block`?

Reply:
Yeah, for now though we can also just check for intersection between targets of `default_block` and `block` and not proceed if there is one.